--- a/arch/arm/boot/dts/allwinner/sun7i-a20-olinuxino-lime2-emmc.dts
+++ b/arch/arm/boot/dts/allwinner/sun7i-a20-olinuxino-lime2-emmc.dts
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 #include "sun7i-a20-olinuxino-lime2.dts"
+#include <dt-bindings/gpio/gpio.h>
 
 / {
 	model = "Olimex A20-OLinuXino-LIME2-eMMC";
@@ -20,3 +21,49 @@
 	status = "okay";
 	non-removable;
 };
+
+/ {
+	panel: panel {
+		compatible = "panel-lvds";
+		width-mm = <410>;
+		height-mm = <230>;
+		data-mapping = "jeida-18";
+		status = "okay";
+
+		panel-timing {
+			clock-frequency = <70000000>;
+			hactive = <1366>;
+			hsync-len = <1>;
+			hfront-porch = <20>;
+			hback-porch = <53>;
+			vactive = <768>;
+			vsync-len = <1>;
+			vfront-porch = <17>;
+			vback-porch = <22>;
+		};
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&tcon0_out_panel>;
+			};
+		};
+	};
+};
+
+&de {
+	status = "okay";
+};
+
+&tcon0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&lvds0_pins>;
+	status = "okay";
+};
+
+&tcon0_out {
+	tcon0_out_panel: endpoint@0 {
+		reg = <0>;
+		remote-endpoint = <&panel_in>;
+	};
+};
