diff --git a/drivers/gpu/drm/sun4i/sun4i_lvds.c b/drivers/gpu/drm/sun4i/sun4i_lvds.c
--- a/drivers/gpu/drm/sun4i/sun4i_lvds.c
+++ b/drivers/gpu/drm/sun4i/sun4i_lvds.c
@@ -99,7 +99,10 @@ int sun4i_lvds_init(struct drm_device *drm, struct sun4i_tcon *tcon)
 	ret = drm_of_find_panel_or_bridge(tcon->dev->of_node, 1, 0,
 					  &lvds->panel, &bridge);
 	if (ret) {
-		dev_info(drm->dev, "No panel or bridge found... RGB output disabled\n");
+		if (ret == -ENODEV)
+			ret = -EPROBE_DEFER;
+		if (ret != -EPROBE_DEFER)
+			dev_info(drm->dev, "No panel or bridge found... RGB output disabled\n");
 		return ret;
 	}
